dockers/docker-database/docker-database-init.sh

```diff
@@ -79,4 +79,18 @@ else
 fi
 rm $db_cfg_file_tmp
 
+# copy dump.rdb file to each instance for restoration
+DUMPFILE=/var/lib/redis/dump.rdb
+redis_inst_list=`/usr/bin/python -c "import swsssdk; print(' '.join(swsssdk.SonicDBConfig.get_instancelist().keys()))"`
+for inst in $redis_inst_list
+do
+    mkdir -p /var/lib/$inst
+    if [[ -f $DUMPFILE ]]; then
+        # copy warmboot rdb file into each new instance location
+        cp $DUMPFILE /var/lib/$inst/dump.rdb
```

Comment:
If `cp` source file is same as the destination, there will be error message and exit code.

Reply:
I tried it on DUT earlier, it just shows a msg telling it is the same file, it didn't affect the function.

Reply:
`cp -n` fits this use case better.

---
In reply to: [536449280](https://github.com/Azure/sonic-buildimage/pull/5773#discussion_r536449280) [](ancestors = 536449280)

Reply:
agree. Updated

Reply:
I revisit the behavior, to not change today’s behavior,  cp -u is correct instead of -n. There is a case we need to overwrite. For example, after warm boot, we do a cold boot, an empty rdb file wil overwrite the previous warm boot rdb file. So -u is correct, only newer source will be copied.

Reply:
Thanks for the insight and sorry for the misleading. Even `cp -u` is not exactly what you truly need, considering the extreme the case the time adjusted backward. So either option is ok:
1. revert back to bare `cp`, and keep error messages
2. check $DUMPFILE != /var/lib/$inst/dump.rdb before `cp`

---
In reply to: [536868596](https://github.com/Azure/sonic-buildimage/pull/5773#discussion_r536868596) [](ancestors = 536868596,536449280)

Reply:
I am checking if src and dst file is the same. Updated